### PR TITLE
Lego 3649 komponenter stotte for custom aria labels pa radio filter stepcontent

### PR DESCRIPTION
**Pull request by vdalseng**
# PR Checklist
https://elvia.atlassian.net/wiki/spaces/TEAMATOM/pages/10427498683/Review+prosess

- [x] Bumpet package?
- [x] Updated changelog?
- [x] Correct date in changelog?

## Describe PR briefly:
Added custom aria-label support on radio filter, stepper and dropdown.

### CODE REVIEW

File: packages/components/components/elvis-dropdown/src/react/dropdown-item/dropdownItem.tsx [hunk 1]
```diff
@@ -156,6 +156,16 @@ export const DropdownItem: React.FC<DropdownItemProps> = ({
...
 
+  const lang = useLanguage();
+  const translatedAriaLabel = () => {
+    return {
+      submenu: lang === 'no' ? 'undermeny' : 'submenu',
+      note: lang === 'no' ? 'Merknad:' : 'Note:',
+    };
+  };
+
+  const labels = translatedAriaLabel();
```
Comment (TrygveUrdahl) [inline packages/components/components/elvis-dropdown/src/react/dropdown-item/dropdownItem.tsx:167]: Any reason it isn't done like this? Why the extra function `translatedAriaLabel`?
```suggestion
  const labels = {
      submenu: lang === 'no' ? 'undermeny' : 'submenu',
      note: lang === 'no' ? 'Merknad:' : 'Note:',
    };```

File: packages/components/components/elvis-stepper/src/react/utils.ts [hunk 2]
```diff
@@ -43,25 +45,47 @@ export const numberShouldBeVisible = (
 };
 
 export const generateStatusMessage = (currentStep: number, steps: StepStates, errorSteps: number[]) => {
-  let statusMessage = `På steg ${currentStep}. `;
+  const lang = useLanguage();
```
Comment (TrygveUrdahl) [inline packages/components/components/elvis-stepper/src/react/utils.ts:48]: You cannot use a hook inside a utility-function. React hooks may only be called in components or other hooks, so this should instead be refactored. 
Perhaps as a parameter to the function instead, and the hook called in the component?

File: packages/components/components/elvis-stepper/src/react/utils.ts [hunk 3]
```diff
@@ -43,25 +45,47 @@ export const numberShouldBeVisible = (
...
+      statusLabel: (isCompleted: boolean) =>
+        isCompleted
+          ? lang === 'no'
+            ? 'vellykket'
+            : 'successful'
+          : lang === 'no'
+            ? 'ikke fullført'
+            : 'not completed',
+    };
+  };
```
Comment (TomSchrier) [inline packages/components/components/elvis-stepper/src/react/utils.ts:66]: I would make this function return two objects; one in English, and one in Norwegian, then, you will also fix the... 

> Extract this nested ternary operation into an independent statement.

...warning that Sonar Cloud is giving.

Maybe something like this:

```
const getTranslatedLabels = () => {
    lang === 'no' ? {
      onStep: 'På steg',
      previousStep: 'Det forrige steget var',
      osv...
    } : {
      onStep: 'On step',
      previousStep: 'The previous step was',
    etc...
    }}
```

File: packages/components/components/elvis-dropdown/src/react/dropdown-item/dropdownItem.tsx [hunk 4]
```diff
@@ -178,8 +188,8 @@ export const DropdownItem: React.FC<DropdownItemProps> = ({
         aria-haspopup={item.children ? 'listbox' : 'false'}
         aria-expanded={isShowing}
         aria-selected={selfOrAllChildrenAreSelected}
-        aria-label={`${item.label}${item.children ? ' , undermeny' : ''}${
-          item.tooltip ? ' , Merknad: ' + item.tooltip : ''
+        aria-label={`${item.label}${item.children ? ` , ${labels.submenu}` : ''}${
+          item.tooltip ? ` , ${labels.note} ${item.tooltip}` : ''
```
Comment (TomSchrier) [inline packages/components/components/elvis-dropdown/src/react/dropdown-item/dropdownItem.tsx:192]: Be aware that this bit gives a "Refactor this code to not use nested template literals." on SonarCloud

File: packages/components/components/elvis-dropdown/src/react/dropdown-overlay/dropdownOverlay.tsx [hunk 5]
```diff
@@ -1,4 +1,5 @@
 import { FormFieldSizes, Overlay } from '@elvia/elvis-toolbox';
+import { LanguageCode } from '@elvia/elvis-toolbox/src';
```
Comment (TomSchrier) [inline packages/components/components/elvis-dropdown/src/react/dropdown-overlay/dropdownOverlay.tsx:2]:
```suggestion
import { FormFieldSizes, Overlay, LanguageCode } from '@elvia/elvis-toolbox';
```

File: packages/components/components/elvis-stepper/src/react/utils.ts (hunk 2 above)
Comment (TrygveUrdahl) [reply packages/components/components/elvis-stepper/src/react/utils.ts:48]: See the note below here: https://react.dev/learn/reusing-logic-with-custom-hooks#hook-names-always-start-with-use
image

File: packages/components/components/elvis-stepper/src/react/utils.ts (hunk 3 above)
Comment (vdalseng) [reply packages/components/components/elvis-stepper/src/react/utils.ts]: This has been implemented.

File: packages/components/components/elvis-stepper/src/react/utils.ts (hunk 2 above)
Comment (vdalseng) [reply packages/components/components/elvis-stepper/src/react/utils.ts]: I have addressed the issue and removed the directly implemented hook in utils.ts. Instead I implemented the hook in both Stepper-horizontal.tsx and vertical.tsx, and passed the lang variable to the utils file as a parameter.

File: packages/components/components/elvis-dropdown/src/react/dropdown-item/dropdownItem.tsx (hunk 1 above)
Comment (vdalseng) [reply packages/components/components/elvis-dropdown/src/react/dropdown-item/dropdownItem.tsx:167]: I made the changes you suggested.

File: packages/components/components/elvis-stepper/src/react/utils.ts [hunk 6]
```diff
@@ -44,32 +44,39 @@ export const numberShouldBeVisible = (
...
+      : {
+          onStep: 'On step',
+          previousStep: 'The previous step was',
+          steps: 'Steps',
+          and: 'and',
+          wasInvalid: 'was invalid',
+          successful: 'successful',
+          notCompleted: 'not completed',
+        };
+  })();
```
Comment (TrygveUrdahl) [inline packages/components/components/elvis-stepper/src/react/utils.ts:73]: There shouldn't be any need for this to be an [IIFE](https://developer.mozilla.org/en-US/docs/Glossary/IIFE), so just convert it to a regular object maybe?

File: packages/components/components/elvis-dropdown/src/react/dropdown-item/dropdownItem.tsx (hunk 4 above)
Comment (vdalseng) [reply packages/components/components/elvis-dropdown/src/react/dropdown-item/dropdownItem.tsx]: Fixed.

**Review by TrygveUrdahl — APPROVED** [review]
LGTM!

**Pull request merged**